### DISCUSSION TITLE
Does inngest support priority queue?

### DISCUSSION THREAD

**Discussion by offchan42** [inngest/inngest#639] [Q&A] [answered]
My use case is to have free users and paid users for AI image generation service and I want paid users to frontrun free users when they enqueue their jobs.

I'm not super sure exactly about which data structure or algorithm I need but roughly I need to be able to rearrange the order of jobs based on user tier and their usage frequency somehow. Paid users will enjoy fast AI image generation and free users will not. Users on the same tier that use the service more often than others also should be deprioritized.
You can suggest me a solution. Thanks.

**Comment by tonyhb** [comment, marked as answer]
Yes, you can read our docs here: https://www.inngest.com/docs/reference/functions/run-priority.

This allows you to initialize function runs with dynamic priorities based off of event data.

**Comment by zmays** [comment]
This only works within the same function.

I'm trying to make a global concurrency queue: https://www.inngest.com/docs/guides/concurrency#how-global-limits-work that would then allow prioritizing certain functions over others. That doesn't seem do-able currently, right? The example linked above mentioned setting different limits in each function, but then you can unnecessarily cap a function.
